cli/flox/src/commands/auth.rs

```diff
@@ -74,25 +79,70 @@ pub async fn authorize(client: BasicClient) -> Result<Credential> {
 
     debug!("Device code details: {details:#?}");
 
-    eprintln!(
-        "Please visit {} in your browser",
-        details.verification_uri().as_str()
-    );
-    eprintln!("And enter code: {}", details.user_code().secret());
+    let opener = Browser::detect();
+
+    let done = Arc::new(AtomicBool::default());
+
+    match opener {
+        Ok(opener) => {
+            let message = formatdoc! {"
+            Verification Code: {code}
+
+            Press [enter] to open {url} in your browser and verify you see the code above.
```

Comment:
blocking (although maybe I've missed :
1. The ticket says to use hub.flox.dev. I'm seeing dev-j4tiszdm1f0b70xf.us.auth0.com, is that cause I'm on the dev instance, or do we need to change that?
    - (nonblocking related nit) personally I don't think we should print something different than we're opening, because I would be confused if I was told one URL was being opened and another actually was. I think we should drop the URL or even just change the wording to `login to hub.flox.dev`. Thoughts @jennymahmoudi ?
2. There's some other discrepancies in the output with the ticket, but particularly this should say to enter a code, not verify you see the code

Reply:
> 1. [..] `dev-j4tiszdm1f0b70xf.us.auth0.com, is that cause I'm on the dev instance, or do we need to change that`

We'll change that I think once we have access to the `flox` tenant at auth0. then i guess it will be something like `flox.us.auth0.com`.

> 1. [..] `login to hub.flox.dev`

I like that, logging in to the actual _hub_ makes more sense as a "vanity url" than the somewhat cryptic auth0 url.

Tbf it was already supposed to be the floxhb url:

> where [vanity URL] is the base url of floxhub: `hub.flox.dev`
> #719

> 2. [..] verify you see the code

ah yes that was from before when we opened the url with the code included..